include/nbl/video/IPhysicalDevice.h

```diff
@@ -30,6 +30,7 @@ class IPhysicalDevice : public core::Interface, public core::Unmovable
             uint32_t SSBOAlignment;
             uint32_t bufferViewAlignment;
             float    maxSamplerAnisotropyLog2;
+            float    timestampPeriodInNanoSeconds;
```

Comment:
why on earth is it a float?

Reply:
because queries are written into uint32_t's, ask Vulkan why 
https://www.khronos.org/registry/vulkan/specs/1.3-extensions/man/html/VkPhysicalDeviceLimits.html
https://www.khronos.org/registry/vulkan/specs/1.3-extensions/html/vkspec.html#queries-timestamps

Reply:
I think I named it bad. and caused a confusion

Reply:
It means that if your difference is `2` then the time period is `2 * timestampPeriod` nanoseconds

Reply:
@devshgraphicsprogramming  timestampPeriod is really in nano seconds. it may be a floating point nanoseconds. it just really what you multiply by a diff number and get a number in nanoseconds. 

https://vulkan.gpuinfo.org/displaydevicelimit.php?name=timestampPeriod&platform=all

Reply:
I'm up for suggestion for a better name for it `timestampPeriod is the number of nanoseconds required for a timestamp query to be incremented by 1`

Reply:
my main beef is that its a `float` which makes no sense (cause you only count WHOLE nanoseconds as the period), so either make it a `uint32_t` and `uint64_t`

Reply:
Vulkan reports float, Do you want me to round it up?
doesn't really matter, most of the devices return 1 anyways. only a handful and weird devices return 0<timestamPeriod<1 and > 1

Reply:
some Intel HD Graphics gpus on arch and ubunto return `83.33333587646484` LOL

Reply:
ok then add a comment about why its a `float` cause using floats in timekeeping is disturbing AF.
add a comment about using a `core::rational<uint32_t>` in the future